### ISSUE TITLE
Change starting number in the collection

### ISSUE THREAD

**Issue by soleaswest** [HashLips/hashlips_art_engine#366]
Hey there,

Does anybody knows how to change the starting number of the collection instead of 0?

I want it to start from 5000 for example and so on

**Comment by debashisdey** [comment]
You cannot do that. What you can do is bulk rename the image and json files with a starting number. But for correcting the json files content you need to run a -sed script in terminal.

**Comment by levianlp** [comment]
@debashisdey thank you for yr advice mate, so do you have any example of a script that can rename the name inside json like for every single json file

**Comment by bolshoytoster** [comment]
@levianlp you can start a collection from any number in  [my repo](https://github.com/bolshoytoster/hashlips_art_engine). You set `startEditionFrom` just like you set `growEditionSizeTo`.

**Comment by levianlp** [comment]
@bolshoytoster ola thank you mate i'll try it now, thank you so much

**Comment by levianlp** [comment]
@bolshoytoster hello i got this error Cannot find module 'D:\hashlips_art_engine-main (1)/node_modules/sha1', what node patch do you use

**Comment by bolshoytoster** [comment]
@levianlp just run `npm install`.

**Comment by levianlp** [comment]
@bolshoytoster yep already do npm install sha1 and it works man, thank you for yr work appreciate it so much

**Comment by Lucobos1970** [comment]
@bolshoytoster Hello, thanks in advance for any help I can get. I am not a developer, but I learned how to make an NFT collection, but I wanted to learn how to start a collection from any number I choose... I got you repo, but I don't know what to do next... I have to set  startEditionFrom  but, where do I do that? in the config.js file ?   I have seen in main.js  code lines containing  startEditionFrom  but, where/how do I set startEditionFrom  in the config.js file?  Thanks a lot!

**Comment by bolshoytoster** [comment]
@Lucobos1970
```js
const layerConfigurations = [
  {
    startEditionFrom: 100, // here
    growEditionSizeTo: 200,
    layersOrder: [
      { name: "Background" },
      { name: "Eyeball" },
      { name: "Eye color" },
      { name: "Iris" },
      { name: "Shine" },
      { name: "Bottom lid" },
      { name: "Top lid" },
    ],
  },
];
```

**Comment by Lucobos1970** [comment]
Hello Bolshoytoster, thanks so much for the fast reply...
I changed the config.js file:

const layerConfigurations = [
  {
    startEditionFrom: 21,
    growEditionSizeTo: 30,
    layersOrder: [
      { name: "Background" },
      { name: "Eyeball" },
      { name: "Eye color" },
      { name: "Iris" },
      { name: "Shine" },
      { name: "Bottom lid" },
      { name: "Top lid" },
    ],
  },
];

But I got this error, it looks like it couldn't find: node_modules/sha1
(But I couldn't find that folder in the repo files)

Error: Cannot find module 'C:\Users\pixel\Desktop\NFT_3D_CHARACTER\3D_CHARACTER_COLLECTIONS\1 - HASHLIPS APP TRIAD\03-hashlips_art_engine-main/node_modules/sha1'
Require stack:
- C:\Users\pixel\Desktop\NFT_3D_CHARACTER\3D_CHARACTER_COLLECTIONS\1 - HASHLIPS APP TRIAD\03-hashlips_art_engine-main\src\main.js
- C:\Users\pixel\Desktop\NFT_3D_CHARACTER\3D_CHARACTER_COLLECTIONS\1 - HASHLIPS APP TRIAD\03-hashlips_art_engine-main\index.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Users\pixel\Desktop\NFT_3D_CHARACTER\3D_CHARACTER_COLLECTIONS\1 - HASHLIPS APP TRIAD\03-hashlips_art_engine-main\src\main.js:5:14)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1155:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'C:\\Users\\pixel\\Desktop\\NFT_3D_CHARACTER\\3D_CHARACTER_COLLECTIONS\\1 - HASHLIPS APP TRIAD\\03-hashlips_art_engine-main\\src\\main.js',
    'C:\\Users\\pixel\\Desktop\\NFT_3D_CHARACTER\\3D_CHARACTER_COLLECTIONS\\1 - HASHLIPS APP TRIAD\\03-hashlips_art_engine-main\\index.js'
  ]
}

**Comment by bolshoytoster** [comment]
@Lucobos1970 you need to run `npm install` again.

**Comment by Lucobos1970** [comment]
Thanks my friend...  sorry to be a pain, but I tried running npm install ...  
In the Build folder, inside the json folder I have all .json files ... but the images folder is empty. 

In the config.js file I tried this code:
// If you have selected Solana then the collection starts from 0 automatically
const layerConfigurations = [
  {
    startEditionFrom: 10,
    growEditionSizeTo: 20,
    layersOrder: [
      { name: "Background" },
      { name: "body" },
      { name: "face" },
      { name: "tatoo" },
      { name: "glasses" },
      { name: "hair" },
      { name: "piercing" },
      { name: "pants" },
      { name: "belt" },
      { name: "shirt" },
      { name: "music" },
      { name: "bracelet" },
      { name: "skateboard" },
      { name: "boots" },
    ],
  },
];


And this is the mssg what I get in the Terminal:

Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.

Try the new cross-platform PowerShell https://aka.ms/pscore6

PS C:\Users\pixel\Desktop\NFT_3D_CHARACTER\3D_CHARACTER_COLLECTIONS\1 - HASHLIPS APP TRIAD\hashlips_art_engine-main> npm install

up to date, audited 67 packages in 3s

6 packages are looking for funding
  run `npm fund` for details      

found 0 vulnerabilities
PS C:\Users\pixel\Desktop\NFT_3D_CHARACTER\3D_CHARACTER_COLLECTIONS\1 - HASHLIPS APP TRIAD\hashlips_art_engine-main> node index.js
Created edition: 10, with DNA: 0373645b7221eac94bd780b69eecfd7127982c05
Created edition: 11, with DNA: bcee5427b33ed75f4381d40cdba4333884c44b40
Created edition: 12, with DNA: f8f136dbdf77a93b534323cb49f887d73e0379c5
Created edition: 13, with DNA: dd29cdb3a89e9376e7bb2985ab608793eb185d39
Created edition: 14, with DNA: b4faee3fdc619052f86066617e9b83d03bbea5f0
Created edition: 15, with DNA: 64c2b014551056251ae6df234077720bb6fd11ee
Created edition: 16, with DNA: f52222dee342a6de20730685793d9628406960bc
Created edition: 17, with DNA: db5ced67874d74c6f862a15520a0c65a763ebe3a
Created edition: 18, with DNA: a2054b3379c5f51e29796d0eee6ba634069a649b
Created edition: 19, with DNA: 850d164b10a8ecb750c296d1057e4c7efd0e9f73
Created edition: 20, with DNA: c4afd355ab44b261834fa586a89ad2687efc8f48
PS C:\Users\pixel\Desktop\NFT_3D_CHARACTER\3D_CHARACTER_COLLECTIONS\1 - HASHLIPS APP TRIAD\hashlips_art_engine-main> 

Thanks

**Comment by bolshoytoster** [comment]
@Lucobos1970 I'm fixing this problem now actually.

**Comment by Lucobos1970** [comment]
Hello bolshoytoster... Thanks really... very very helpful...

**Comment by Gireeshbd** [comment]
Hello @bolshoytoster  and @Lucobos1970  
```
const layerConfigurations = [
  {
    startEditionFrom: 100,
    growEditionSizeTo: 10,
    layersOrder: [
      { name: "Background" },
      { name: "Eyeball" },
      { name: "Eye color" },
      { name: "Iris" },
      { name: "Shine" },
      { name: "Bottom lid" },
      { name: "Top lid" },
    ],
  },
];
```
I had set startEditionFrom : 10 , but still its starting from 1
can you guys help me out in this

**Comment by bolshoytoster** [comment]
@Gireeshbd
1. Are you using [my fork](https://github.com/bolshoytoster/hashlips_art_engine)?
2. If so, have you made sure to save `src/config.js` after editing it?
